python/action-server/main.py

```diff
@@ -0,0 +1,154 @@
+import time
+import json
+import itertools
+import zenoh
+from zenoh import Reliability, SampleKind, Query, Sample, KeyExpr, QueryTarget, Value
+from pydantic import BaseModel
+import logging
+import asyncio
+import yaml
+import state_machine
+log = logging.getLogger(__name__)
+
+# describe settings for parsing values.
+class ActionSettings(BaseModel):
+    mode: str = None
+    connect: str = None
+    listen: str = None
+    config: str = None
+    base_key_expr: str
+    iter: int
+    target: str
+    declare_key_expr: str
+
+store = {}
+
+class Handlers():
+    # handler functions which are going to use session declarables.
+    # listens all the data which publisher puts on reciever.
+    def listener(self, sample: Sample):
+        print(">> [Subscriber] Received {} ('{}': '{}')"
+            .format(sample.kind, sample.key_expr, sample.payload.decode("utf-8")))
+        if sample.kind == SampleKind.DELETE():
+            store.pop(sample.key_expr, None)
+        else:
+            store[sample.key_expr] = sample
+
+    # replies of every query which get function asks.
+    def query_handler(self, query: Query):
+        print(">> [Queryable ] Received Query '{}'".format(query.selector))
+        replies = []
+        for stored_name, sample in store.items():
+            if query.key_expr.intersects(stored_name):
+                query.reply(sample)
+
+    # for publishing the data on subscriber or can be said that the to take the feedback
+    def publisher(self, key, pub, iter):
+        for idx in itertools.count() if iter is None else range(int(iter)):
+            time.sleep(1)
+            buf = (idx % 100) 
+            print(f"Putting Data ('{key}': '{buf}')...")
+            pub.put(buf)
+        
+
+class Session(Handlers):
+    # creates self.session and perform self.session related tasks. Takes an object of settings.
+    def __init__(self, settings):
+        self.setting = settings
+        self.session = None
+        self.sub = None
+        self.pub = None
+        self.queryable = None
+
+    # configures the zenoh configuration from settings variables and returns a configuration object.
+    def configuration(self):
+        conf = zenoh.Config.from_file(
+            self.setting.config) if self.setting.config is not None else zenoh.Config()
+        if self.setting.mode is not None:
+            conf.insert_json5(zenoh.config.MODE_KEY, json.dumps(self.setting.mode))
```

Comment:
configuration management of this sort can be simplified using `Pydantic` and libraries such as `Confz`